src/main/java/com/redislabs/redisgraph/impl/api/ContextedRedisGraph.java

```diff
@@ -0,0 +1,121 @@
+package com.redislabs.redisgraph.impl.api;
+
+import com.redislabs.redisgraph.RedisGraphContexted;
+import com.redislabs.redisgraph.ResultSet;
+import com.redislabs.redisgraph.impl.graph_cache.RedisGraphCaches;
+import com.redislabs.redisgraph.impl.resultset.ResultSetImpl;
+import redis.clients.jedis.Client;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.util.SafeEncoder;
+
+import java.util.List;
+
+/**
+ * An implementaion of RedisGraphContexted. Allows sending RedisGraph and some Redis commands,
+ * within a specific connection context
+ */
+public class ContextedRedisGraph extends AbstractRedisGraph implements RedisGraphContexted, RedisGraphCacheHolder {
+
+    private Jedis connectionContext;
+    private RedisGraphCaches caches;
+
+    /**
+     * Generates a new instance with a specific Jedis connection
+     * @param connectionContext
+     */
+    public ContextedRedisGraph(Jedis connectionContext){
+        this.connectionContext = connectionContext;
+    }
+
+    /**
+     * Overrides the abstract method. Return the instance only connection
+     * @return
+     */
+    @Override
+    protected Jedis getConnection() {
+        return this.connectionContext;
+    }
+
+    /**
+     * Sends the query over the instance only connection
+     * @param graphId graph to be queried
+     * @param preparedQuery prepared query
+     * @return Result set with the query answer
+     */
+    @Override
+    protected ResultSet sendQuery(String graphId, String preparedQuery) {
+        Jedis conn = getConnection();
```

Comment:
try resource block?

Reply:
just try, no need to close resource at the end of the scope